app/controllers/calendar_controller.rb

```diff
@@ -1,11 +1,21 @@
 class CalendarController < ApplicationController
 
   def index
-    @days = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri']
+    week_start = params[:date] ? Date.parse(params[:date]).at_beginning_of_week : Date.today.beginning_of_week
+    week_end = week_start + CalendarHelper::WEEK_LENGTH - 1
+    @days = (week_start..week_end).to_a
+
     start_time = Time.now.at_beginning_of_day
     end_time = Time.now.at_end_of_day
     step = 30.minutes
```

Comment:
if a variable is used once it's ok to do use it inline:
`@times = time_interval(start_time, end_time, 30.minutes)`